corehq/apps/registration/utils.py

```diff
@@ -131,17 +132,38 @@ def request_new_domain(request, form, is_new_user=True):
 
     if is_new_user:
         dom_req.save()
-        send_domain_registration_email(request.user.email,
-                                       dom_req.domain,
-                                       dom_req.activation_guid,
-                                       request.user.get_full_name(),
-                                       request.user.first_name)
+        if settings.IS_SAAS_ENVIRONMENT:
+            load_appcues_template_apps.apply_async(
+                (new_domain.name, current_user.username),
+                link=send_domain_registration_email(
+                    request.user.email,
+                    dom_req.domain,
+                    dom_req.activation_guid,
+                    request.user.get_full_name(),
+                    request.user.first_name
+                )
+            )
+        else:
+            send_domain_registration_email(request.user.email,
+                                           dom_req.domain,
+                                           dom_req.activation_guid,
+                                           request.user.get_full_name(),
+                                           request.user.first_name)
     send_new_request_update_email(request.user, get_ip(request), new_domain.name, is_new_user=is_new_user)
 
     send_hubspot_form(HUBSPOT_CREATED_NEW_PROJECT_SPACE_FORM_ID, request)
     return new_domain.name
 
 
+APPCUES_TEMPLATE_SLUGS = ['appcues']  # TODO: Add agg and wash
+
+@task(serializer='pickle', queue='background_queue')
+def load_appcues_template_apps(domain, username):
+    from corehq.apps.app_manager.views.apps import load_app_from_slug
+    for app_slug in APPCUES_TEMPLATE_SLUGS:
```

Comment:
Are you saying that I should consider putting each app on its own task, or I should use one task for all three? (I'm asking because I thought that's what I was already doing, and I'm confused by the phrasing).

Reply:
I'm saying you should consider putting each app in its own task - sorry, I was being vague  because it's not definitely a good idea. The tradeoff is the time you'd save by parallelizing the tasks, versus the extra code complexity of waiting for all of the tasks to finish. It probably doesn't make much difference, just curious if you'd thought about it.

Reply:
Hmm...mulling this over some more, given that it's only three jobs, let's forget about this unless during testing you find that it takes a long time for the email to get sent. Preemptively adding scope to this PR probably isn't a good idea.

FWIW, it's possible to wait for an arbitrary number of jobs with [gevent](http://www.gevent.org/intro.html#example). HQ does this in a couple of places, like [fetching the staging.yml branches when rebuilding staging](https://github.com/dimagi/commcare-hq/blob/master/scripts/rebuildstaging.py#L81-L107).

Reply:
That is really cool and I will definitely revisit this later!